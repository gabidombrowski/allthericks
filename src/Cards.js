import React from "react";
import { useQuery } from "@apollo/react-hooks";
import { gql } from "apollo-boost";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

const RICKS = gql`
  {
    characters(page: 1, filter: { name: "rick" }) {
      info {
        count
      }
      results {
        image
        name
        status
        species
        type
        origin {
          name
        }
      }
    }
  }
`;

const Cards = () => {
  const { loading, error, data } = useQuery(RICKS);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return data.characters.results.map((character) => (
    <Card className="Card">
      <CardContent className="CardContent">
        <img alt={character.name} src={character.image} />
        <Typography>
          <span className="Stat">Name: </span>
          {character.name}
        </Typography>
        <Typography>
          <span className="Stat">Status: </span>
          {character.status}
        </Typography>
        <Typography>
          <span className="Stat">Species: </span>
          {character.species}
        </Typography>
        <Typography>
          <span className="Stat">Type: </span>
          {character.type}
        </Typography>
        <Typography>
          <span className="Stat">Origin: </span>
          {character.origin.name}
        </Typography>
      </CardContent>
    </Card>
  ));
};

export default Cards;
