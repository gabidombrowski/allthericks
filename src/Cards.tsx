import React, { Fragment } from 'react'
import { useQuery } from '@apollo/react-hooks'
import { gql } from 'apollo-boost'
import { Box, Card, CardContent, CircularProgress, Typography } from '@material-ui/core/'
import './App.css'

const RICKS_QUERY = gql`
  {
    characters(page: 1, filter: { name: "rick" }) {
      info {
        count
      }
      results {
        image
        name
        species
        type
        origin {
          name
        }
      }
    }
  }
`

const Cards = () => {
  const { loading, error, data } = useQuery(RICKS_QUERY)

  if (loading) return (
    <Box justifyContent={'center'} m={3}>
      <CircularProgress />
      <Typography>Loading...</Typography>
    </Box>
  )
  if (error) return <Typography>Error :(</Typography>
  if (!data) return <Typography>No data available...</Typography>

  const cards = (data && data.characters.results.map((character: any, index: any) => (
    <Card key={index} className='Card'>
      <CardContent className='CardContent'>
        <img alt={character.name} src={character.image} />
        <Typography>
          <span className='Stat'>Name: </span>
          {character.name}
        </Typography>
        <Typography>
          <span className='Stat'>Species: </span>
          {character.species}
        </Typography>
        <Typography>
          <span className='Stat'>Type: </span>
          {character.type}
        </Typography>
        <Typography>
          <span className='Stat'>Origin: </span>
          {character.origin.name}
        </Typography>
        {/* <Typography>
          <span className='Stat'>Status: </span>
          {character.status}
        </Typography> */}
      </CardContent>
    </Card>)
  ))

  return (
    <Fragment>
      <Box>Total: {data.characters.info.count}</Box>
      <Box display={'inline-flex'} flexWrap={'wrap'} justifyContent={'center'} m={3}>{cards}</Box>
    </Fragment>
  )
}

export default Cards
