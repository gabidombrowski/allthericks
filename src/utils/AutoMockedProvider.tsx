import React, { ReactNode } from 'react'
//import { gql } from 'apollo-boost'
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { SchemaLink } from 'apollo-link-schema'
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools'
import { printSchema, buildClientSchema } from 'graphql'
import introspectionResult from './schema.json'
import { CustomResolver } from "./mergeResolvers"

const AutoMockedProvider: React.FunctionComponent<{
  children: ReactNode
  mockResolvers?: CustomResolver
}> = ({ children, mockResolvers }) => {
  const schemaSDL = printSchema(
    buildClientSchema({ __schema: introspectionResult.__schema as any }),
  )

  const schema = makeExecutableSchema({
    typeDefs: schemaSDL,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
  })

  addMockFunctionsToSchema({
    schema,
    mocks: mockResolvers,
  })

  const client = new ApolloClient({
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })

  //  Useful for testing:

  //   const RICKS_QUERY = gql`
  //   {
  //     characters(page: 1, filter: { name: "rick" }) {
  //       info {
  //         count
  //       }
  //       results {
  //         image
  //         name
  //         species
  //         type
  //         origin {
  //           name
  //         }
  //       }
  //     }
  //   }
  // `

  //   client.query({ query: RICKS_QUERY }).then(result => {
  //     console.log('query: ', RICKS_QUERY)
  //     console.log('result: ', result)
  //   })

  return <ApolloProvider client={client}>{children}</ApolloProvider>
}

export default AutoMockedProvider