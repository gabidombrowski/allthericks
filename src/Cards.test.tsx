import React from "react";
import renderer, { act } from "react-test-renderer";
import wait from "waait";
import Cards from "./Cards";
import AutoMockedProvider from "utils/AutoMockedProvider";

it("renders", async () => {
  const tree = renderer.create(
    <AutoMockedProvider>
      <Cards />
    </AutoMockedProvider>
  );

  await act(() => wait(0));

  expect(tree).toMatchSnapshot();
});
