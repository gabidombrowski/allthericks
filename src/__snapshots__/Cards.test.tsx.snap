// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    Total: 
    -53
  </div>,
  <div
    className="MuiBox-root MuiBox-root-45"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root Card MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root CardContent"
      >
        <img
          alt="Hello World"
          src="Hello World"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Name: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Species: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Type: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Origin: 
          </span>
          Hello World
        </p>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root Card MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root CardContent"
      >
        <img
          alt="Hello World"
          src="Hello World"
        />
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Name: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Species: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Type: 
          </span>
          Hello World
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Origin: 
          </span>
          Hello World
        </p>
      </div>
    </div>
  </div>,
]
`;
